joblib/parallel.py

```diff
@@ -60,13 +66,35 @@
 
 DEFAULT_THREAD_BACKEND = 'threading'
 
+# Backend hints and constraints to help choose the backend
```

Comment:
Except for the `TASK_*` variables definition, the diff here is somewhat unrelated to the PR. Maybe address independently of #588 ?